src/iam/iam-base.ts

```diff
@@ -131,7 +131,8 @@ export class IAMBase {
     natsServerUrl,
     bridgeUrl = "https://walletconnect.energyweb.org",
     privateKey,
-    ewKeyManagerUrl = "https://km.aws.energyweb.org/connect/new"
+    ewKeyManagerUrl = "https://km.aws.energyweb.org/connect/new",
+    cacheClient
```

Comment:
I'm a little confused on whether or not we should we adding back the `cacheClient` in the constructor. I think @JGiter and I discussed this and we settled on using the `setCacheClientOptions` function that is in `chainConfig.ts`https://github.com/energywebfoundation/iam-client-lib/pull/124#discussion_r582541327 . So I think if we add it back, then we will have several ways of setting the `cacheClient` config. Maybe this is okay but maybe it is better to have one way. @JGiter Any thoughts?

Reply:
ahhh ok 👍  I'll change it back, but still we need to update the readme ;)

Reply:
@jrhender @JGiter but you need to remember that in this implementation the iam-client-lib user is not able to pass his own cacheServerClient implementation, they need to use our one.

Reply:
@dwojno That's a great point, I hadn't thought of that tbh 😅 . How about having an optional parameter that is a factory function: `cacheClientFactory?: (CacheServerClientOptions) => ICacheServerClient`. In this way, we still rely on the `setCacheClientOptions` to set the cache-server config, but also give the option to override the implementation. What do you think?